src/WaitosaurHolder.sol

```diff
@@ -1,77 +1,49 @@
 pragma solidity ^0.8.28;
 
-import {
-    Initializable
-} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
-import {
-    UUPSUpgradeable
-} from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
-import {
-    OwnableUpgradeable
-} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
-import {StorageSlot} from "@openzeppelin/contracts/utils/StorageSlot.sol";
 import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
-import {
-    SafeERC20
-} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
-
-contract WaitosaurHolder is Initializable, UUPSUpgradeable, OwnableUpgradeable {
-    struct WaitosaurConfig {
-        address token;
-        address locker;
-        address unLocker;
-        address receiver;
-    }
-    struct WaitosaurState {
-        uint256 lockedAmount;
-        uint256 lastLocked;
-    }
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {WaitosaurBase, WaitosaurState} from "./WaitosaurBase.sol";
+
+struct WaitosaurHolderConfig {
+    address token;
+    address receiver;
+}
+
+contract WaitosaurHolder is WaitosaurBase {
+    // ---------------------------------------------------------------------
+    // Errors
+    // ---------------------------------------------------------------------
 
-    // Custom errors
     error InvalidTokenAddress();
     error InvalidReceiverAddress();
-    error InvalidLockerAddress();
-    error InvalidUnlockerAddress();
-    error AlreadyUnlocked();
-    error AlreadyLocked();
-    error AmountZero();
-    error NothingLocked();
-    error InsufficientBalance();
     error NotLocker();
     error NotUnLocker();
 
-    /// @dev keccak256(abi.encode(uint256(keccak256("maxbtc.waitosaur.config")) - 1)) & ~bytes32(uint256(0xff))
-    bytes32 private constant CONFIG_STORAGE_SLOT =
-        0x8f3661154de8aca85dd40a286ee737a27a3b740a5d65a00dc9530f3fcf760200;
-    /// @dev keccak256(abi.encode(uint256(keccak256("maxbtc.waitosaur.state")) - 1)) & ~bytes32(uint256(0xff))
-    bytes32 private constant STATE_STORAGE_SLOT =
-        0xcb11509986cbc674883f88160c41f636c73566a871549bf8441ad8bc0e648300;
+    // ---------------------------------------------------------------------
+    // Events
+    // ---------------------------------------------------------------------
+
+    event ConfigUpdated(WaitosaurHolderConfig config);
+
+    // ---------------------------------------------------------------------
+    // Slots
+    // ---------------------------------------------------------------------
 
-    event Locked(uint256 amount);
-    event Unlocked();
-    event ConfigUpdated(WaitosaurConfig config);
+    /// @dev keccak256(abi.encode(uint256(keccak256("maxbtc.waitosaur.holder.config")) - 1)) & ~bytes32(uint256(0xff))
+    bytes32 private constant CONFIG_STORAGE_SLOT =
+        0xd788cf889637414408102ead9bb45ec368e1e6e80911edf1852af5b5d59ad900;
```

Comment:
This comment and this slot are different.

Reply:
hmm, are you sure?